dockstore-integration-testing/src/test/java/io/dockstore/client/cli/OrganizationIT.java

```diff
@@ -1683,6 +1683,35 @@ public void testDuplicateAliasOperations() {
         }
     }
 
+    /**
+     * Tests that we are getting the number of workflows and tools correctly
+     */
+    @Test
+    public void testWorkflowsToolsLength() {
+        // Setup user who creates Organization and collection
+        final ApiClient webClientUser2 = getWebClient(USER_2_USERNAME, testingPostgres);
+        OrganizationsApi organizationsApi = new OrganizationsApi(webClientUser2);
+
+        // Create the Organization and collection
+        Organization organization = createOrg(organizationsApi);
+        Collection stubCollection = stubCollectionObject();
+
+        long wLength = 2;
+        long tLength = 3;
+        stubCollection.setWorkflowsLength(wLength);
+        stubCollection.setToolsLength(tLength);
+
+        // Attach collections
+        Collection collection = organizationsApi.createCollection(organization.getId(), stubCollection);
+
+        // Approve the org
+        testingPostgres.runUpdateStatement(
+                "update organization set status = '" + io.dockstore.webservice.core.Organization.ApplicationState.APPROVED.toString() + "'");
+
+        assertEquals(wLength, (long)collection.getWorkflowsLength());
```

Comment:
Not sure I understand this test. `stubCollectionObject` creates an empty collection, lines 1701 and 1702 set the length, but the length should be transient. Why are lines 1711 and 1712 passing?

Reply:
@denis-yuen  Yeah line 1711 is failing, I assume they have not yet checked line 1712. Since `stubCollectionObject` is an empty collection, then in order to make the length work I should add tools and workflows to the collection and then retrieve the length?

Reply:
Yeah, I think you basically need to really add tools and workflows to the collection to make the test work